### PR TITLE
Use debounced function to make new search request

### PR DESCRIPTION
**Pull request by kiranparajuli589**
### Description
With this PR: `makeSearchRequest` function is debounced 500 ms to make a new search request.

### Related
- OP#41514
- https://community.openproject.org/projects/nextcloud-integration/work_packages/41514

### CODE REVIEW

File: src/components/tab/SearchInput.vue [hunk 1]
```diff
@@ -110,6 +111,17 @@ export default {
 				this.state = STATE_ERROR
 			}
 		},
+		async asyncFind(query) {
+			query = query.trim()
```
Comment (individual-it) [inline src/components/tab/SearchInput.vue:115]: do we want to search for things like `new `?

File: tests/jest/components/tab/SearchInput.spec.js [hunk 2]
```diff
@@ -159,17 +162,18 @@ describe('SearchInput.vue tests', () => {
...
 				})
-				const searchList = wrapper.find(searchListSelector)
-
-				expect(searchList.exists()).toBeTruthy()
-				expect(searchList).toMatchSnapshot()
+				const multiSelectContent = wrapper.find(multiSelectContentSelector)
+				expect(multiSelectContent.exists()).toBeTruthy()
+				const workPackages = multiSelectContent.findAll(workPackageStubSelector)
+				expect(workPackages).toHaveLength(workPackagesSearchResponse.length)
+				expect(workPackages.at(0).props()).toMatchSnapshot()
```
Comment (individual-it) [inline tests/jest/components/tab/SearchInput.spec.js:176]: why checking only one and not all?

File: src/components/tab/SearchInput.vue [hunk 3]
```diff
@@ -110,6 +111,17 @@ export default {
...
+			if (query) {
+				this.resetState()
+				if (query.length <= SEARCH_CHAR_LIMIT) return
+				await this.debounceMakeSearchRequest(query)
+			}
+		},
+		debounceMakeSearchRequest: _.debounce(function(...args) {
+			return this.makeSearchRequest(...args)
+		}, 500),
```
Comment (individual-it) [inline src/components/tab/SearchInput.vue:124]: put the number in some const?

**Review by individual-it — CHANGES_REQUESTED** [review]
small comments
but it seems not to work for me
image

File: src/components/tab/SearchInput.vue (hunk 3 above)
Comment (kiranparajuli589) [reply src/components/tab/SearchInput.vue]: done

File: tests/jest/components/tab/SearchInput.spec.js (hunk 2 above)
Comment (kiranparajuli589) [reply tests/jest/components/tab/SearchInput.spec.js]: fixed

File: src/components/tab/SearchInput.vue (hunk 1 above)
Comment (kiranparajuli589) [reply src/components/tab/SearchInput.vue]: trim removed

**Comment by kiranparajuli589** [comment]
> small comments but it seems not to work for me image

@individual-it it works for me, can you please check again. I've checked with node v14 and v16 and it works for both. I'm using npm 8.5.0

**Comment by individual-it** [comment]
something went wrong with CI, restarted

**Review by individual-it — APPROVED** [review]
works now.
Any way we could catch that kind of issue with tests, so that it would not happen again?

**Comment by kiranparajuli589** [comment]
> works now. Any way we could catch that kind of issue with tests, so that it would not happen again?

This was a javascript error. occ app enable/disable was working fine. I think we have to open the UI to check this. acceptance would have catched this problem.

**Pull request merged**